inc/dg/CMakeLists.txt

```diff
@@ -0,0 +1,327 @@
+  topology/refined_grid.h
+  topology/refined_gridX.h
+  topology/split_and_join.h
+  topology/stencil.h
+  topology/tensor.h
+  topology/transform.h
+  topology/weights.h
+  topology/weightsX.h
+  topology/xspacelib.h
+)
+
+set(DG_BENCHMARKS
+  arakawa_b.cpp
+  blas_b.cpp
+  cg2d_b.cpp
+  cluster_b.cpp
+  elliptic2d_b.cpp
+  ellipticX2d_b.cpp
+  elliptic_b.cpp
+  helmholtz_b.cpp
+  helmholtzg2_b.cpp
+  multigrid_b.cpp
+  poisson_b.cpp
+  refined_elliptic2d_b.cpp
+)
+
+set(DG_TESTS
+  adaptive_t.cpp
+  arakawa_t.cpp
+  backend/fma_t.cpp
+  backend/index_t.cpp
+  backend/memory_t.cpp
+  backend/sparsematrix_t.cpp
+  backend/traits_t.cpp
+  backend/view_t.cpp
+  bicgstabl_t.cpp
+  blas1_t.cpp
+  blas_t.cpp
+  eve_t.cpp
+  extrapolation_t.cpp
+  helmholtz_t.cpp
+  implicit_t.cpp
+  multistep_t.cpp
+  nullstelle_t.cpp
+  pcg_t.cpp
+  poisson_t.cpp
+  runge_kutta_t.cpp
+  simpsons_t.cpp
+  topology/average_t.cpp
+  topology/base_geometry_t.cpp
+  topology/derivativesX_t.cpp
+  topology/derivatives_t.cpp
+  topology/dlt_t.cpp
+  topology/dx_t.cpp
+  topology/evaluation_t.cpp
+  topology/fem_t.cpp
+  topology/filter_t.cpp
+  topology/geometry_t.cpp
+  topology/grid_t.cpp
+  topology/interpolation_t.cpp
+  topology/multiply_t.cpp
+  topology/operator_t.cpp
+  topology/operator_tensor_t.cpp
+  topology/projection_t.cpp
+  topology/refined_grid_t.cpp
+  topology/stencil_t.cpp
+  topology/tensor_t.cpp
+  topology/transform_t.cpp
+)
+
+set(DG_MPI_TESTS
+  backend/mpi_exblas_mpit.cpp
+  backend/mpi_gather_kron_mpit.cpp
+  backend/mpi_gather_mpit.cpp
+  backend/mpi_init_mpit.cpp
+  backend/mpi_kron_mpit.cpp
+  backend/mpi_permutation_mpit.cpp
+  topology/filter_mpit.cpp
+  topology/interpolation_mpit.cpp
+  topology/mpi_grid_mpit.cpp
+  topology/stencil_mpit.cpp
+)
+
+# ============================================
+# Create DG library target
+
+# Create the dg target library.
+# As a header-only library, this must be an interface library.
+# This means it has no sources, but can link/be linked to other
+# libraries, have dependencies, etc. These properties are then
+# inherited by any target that links to it.
+add_library(dg INTERFACE ${DG_HEADERS})
+# Ensure that `#include "dg/..."` works
+# TODO remove this after other libs have their own CMakeLists.txt
+target_include_directories(dg INTERFACE "${PROJECT_SOURCE_DIR}/inc")
+# Set the minimum C++ standard
+target_compile_features(dg INTERFACE cxx_std_17)
+# Set flags
+# CMake switches on -forward-unknown-to-host-compiler by default, so C++ flags
+# may be set even when compiling for CUDA.
+target_compile_options(dg INTERFACE
+  "${FELTOR_FLAGS_CXX}"
+  $<$<COMPILE_LANGUAGE:CUDA>:${FELTOR_FLAGS_CUDA}>
+)
+
+# Create an alias: used to link to the library from external projects.
+add_library(dg::dg ALIAS dg)
+
+# ============================================
+# Add dg dependencies
+
+include(FeltorFetchDeps)
+
+if(FELTOR_FETCH_CCCL)
+  fetch_cccl()
+else()
+  find_package(cccl REQUIRED)
+endif()
```

Comment:
So, I have raised an issue https://github.com/NVIDIA/cccl/issues/4452 with the cccl library, to tell us exactly how we are supposed to link the targets, because their documentation is unclear. It appears we do not manually have to define the `THRUST_DEVICE_SYSTEM` Macros and I would expect the `FELTOR_USE_OMP` and `FELTOR_USE_GPU` options to be replaced by thrust options. Let's see what they come back with